### PR TITLE
Work with entities and repos

### PR DESCRIPTION
**Pull request by BlazeOfSoul**

### CODE REVIEW

File: server/RafRate.API/Controllers/ActiveDayController.cs [hunk 1]
```diff
@@ -10,8 +10,5 @@ public class ActiveDayController : ControllerBase
 {
     private readonly IActiveDayService _activeDayService;
 
-    public ActiveDayController(ActiveDayService activeDayService)
-    {
-        _activeDayService = activeDayService;
-    }
-}
\ No newline at end of file
+    public ActiveDayController(ActiveDayService activeDayService) => this._activeDayService = activeDayService;
```
Comment (yarmoliq) [inline server/RafRate.API/Controllers/ActiveDayController.cs]: Revert this.
What happens when you add a new dependency?

File: server/RafRate.API/Controllers/AuthenticationController.cs [hunk 2]
```diff
@@ -5,26 +5,28 @@
 
 namespace RafRate.API.Controllers;
 
+using Data.Entities.Users;
```
Comment (yarmoliq) [inline server/RafRate.API/Controllers/AuthenticationController.cs]: Why is this `using` separated from other usings?

File: server/RafRate.API/Controllers/CommentController.cs [hunk 3]
```diff
@@ -10,8 +10,5 @@ public class CommentController : ControllerBase
 {
     private readonly ICommentService _commentService;
 
-    public CommentController(CommentService commentService)
-    {
-        _commentService = commentService;
-    }
-}
\ No newline at end of file
+    public CommentController(CommentService commentService) => this._commentService = commentService;
```
Comment (yarmoliq) [inline server/RafRate.API/Controllers/CommentController.cs]: Revert

File: server/RafRate.API/Controllers/TeacherController.cs [hunk 4]
```diff
@@ -10,8 +10,5 @@ public class TeacherController : ControllerBase
 {
     private readonly ITeacherService _teacherService;
 
-    public TeacherController(TeacherService teacherService)
-    {
-        _teacherService = teacherService;
-    }
-}
\ No newline at end of file
+    public TeacherController(TeacherService teacherService) => this._teacherService = teacherService;
```
Comment (yarmoliq) [inline server/RafRate.API/Controllers/TeacherController.cs]: Revert

File: server/RafRate.Data/Entities/Teachers/TeacherEntity.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+using RafRate.Data.Interfaces;
+
+public class TeacherEntity : IEntity
+{
+    public Guid Id { get; set; }
+
+    public string Name { get; set; }
+
+    public string JobTittle { get; set; }
```
Comment (yarmoliq) [inline server/RafRate.Data/Entities/Teachers/TeacherEntity.cs]: Title misspelled

File: server/RafRate.Data/Interfaces/IBaseRepository.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace RafRate.Data.Interfaces;
+
+public interface IBaseRepository<TEntity>
+{
+    Task<TEntity> CreateAsync(TEntity entity);
+
+    Task<TEntity> UpdateAsync(TEntity entity);
+
+    Task<TEntity> GetByAsync(Expression<Func<TEntity, bool>> expression, params Expression<Func<TEntity, object>>[] includes);
```
Comment (yarmoliq) [inline server/RafRate.Data/Interfaces/IBaseRepository.cs]: Дима, ты ему подсунул это?

File: server/RafRate.Data/Repositories/Base/BaseRepository.cs [hunk 7]
```diff
@@ -3,12 +3,48 @@
...
+
     public BaseRepository(DbContext context)
     {
         _context = context;
+        _dbSet = context.Set<TEntity>();
+    }
+
+    public async Task<TEntity> CreateAsync(TEntity entity)
+    {
+        var newEntity = await _context.AddAsync(entity);
```
Comment (yarmoliq) [inline server/RafRate.Data/Repositories/Base/BaseRepository.cs]: Should probably rename `newEntity` to smth like `entry`

File: server/RafRate.Data/Interfaces/IBaseRepository.cs (hunk 6 above)
Comment (yarmoliq) [reply server/RafRate.Data/Interfaces/IBaseRepository.cs]: Надо это обсудить, важно

**Review by yarmoliq — CHANGES_REQUESTED** [review]

File: server/RafRate.Data/Interfaces/IBaseRepository.cs (hunk 6 above)
Comment (BlazeOfSoul) [reply server/RafRate.Data/Interfaces/IBaseRepository.cs]: Please speak english i don't understand hahahhaha

File: server/RafRate.API/Controllers/AuthenticationController.cs [hunk 8]
```diff
@@ -1,30 +1,32 @@
...
 public class AuthenticationController : ControllerBase
 {
     private readonly IAuthenticationService _authenticationService;
     private readonly AppDbContext _dbContext;
+
     public AuthenticationController(IAuthenticationService authenticateService, AppDbContext dbContext)
     {
-        _authenticationService = authenticateService;
-        _dbContext = dbContext;
+        this._authenticationService = authenticateService;
```
Comment (bogomazdmitry) [inline server/RafRate.API/Controllers/AuthenticationController.cs]: `This` is not required

File: server/RafRate.API/Program.cs [hunk 9]
```diff
@@ -9,7 +9,7 @@
 builder.Services.AddControllers();
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
-builder.Services.AddCors(options => options.AddPolicy(name: "RafRateOrigins",
+builder.Services.AddCors(options => options.AddPolicy("RafRateOrigins",
     policy =>
     {
         policy.WithOrigins("http://localhost:4200").AllowAnyMethod().AllowAnyHeader();
```
Comment (bogomazdmitry) [inline server/RafRate.API/Program.cs]: I think localhost url is not ok

File: server/RafRate.Data/Entities/ActiveDays/ActiveDayEntity.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
+﻿namespace RafRate.Data.Entities.ActiveDays;
+
+using RafRate.Data.Interfaces;
+
+public class ActiveDayEntity : IEntity
```
Comment (bogomazdmitry) [inline server/RafRate.Data/Entities/ActiveDays/ActiveDayEntity.cs:5]: If you use Guid in every entity I think you can move id property to the base class

File: server/RafRate.Data/Interfaces/IBaseRepository.cs (hunk 6 above)
Comment (bogomazdmitry) [reply server/RafRate.Data/Interfaces/IBaseRepository.cs]: https://metanit.com/sharp/entityframeworkcore/3.9.php

**Review by bogomazdmitry — CHANGES_REQUESTED** [review]

**Pull request merged**